### PR TITLE
LPD-45136 Only delete the user if they haven't already been deleted

### PR DESCRIPTION
**Pull request by brookedalton**
https://liferay.atlassian.net/browse/LPD-45136

### CODE REVIEW

**Comment by brookedalton** [comment]
ci:test:publications

File: modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/spi/listener/CTOnDemandUserEventListener.java [hunk 1]
```diff
@@ -62,7 +62,11 @@ private void _deleteOnDemandUser(CTCollection ctCollection)
 				_ticketLocalService.deleteTicket(ticket);
 			}
 
-			_userLocalService.deleteUser(ctCollection.getOnDemandUserId());
+			if (_userLocalService.fetchUser(ctCollection.getOnDemandUserId()) !=
```
Comment (david-truong) [inline modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/spi/listener/CTOnDemandUserEventListener.java:65]: can you assign it to a variable instead and then pass that value into the delete?

**Comment by brookedalton** [comment]
ci:test:sf

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
ci:forward:force

**Comment by brookedalton** [comment]
ci:forward:force

**Pull request closed without merging**